model/src/common/search/index.ts

```diff
@@ -0,0 +1,17 @@
+import Joi from 'joi'
+
+import { type SearchOptions } from '~/src/common/search/types.js'
+
+/**
+ * Field definitions for search options.
+ */
+export const searchOptionFields = {
+  title: Joi.string().trim().min(1).optional()
+}
+
+/**
+ * Joi schema for `SearchOptions` interface
+ * @see {@link SearchOptions}
```

Comment:
Can stick the link in the description if you like?

It lets you do Markdown style link text `{@link SearchOptions | Link text here}` too

```suggestion
 * Joi schema for {@link SearchOptions} interface
```

Reply:
That's more concise for sure.